modules/ai/assets/js/editor/pages/form-layout/hooks/use-screenshots.js

```diff
@@ -53,7 +53,20 @@ const useScreenshots = ( { onData } ) => {
 		const promises = screenshotsData.map( ( { generate } ) => {
 			const prevGeneratedIds = screenshots.map( ( screenshot ) => screenshot.baseTemplateId );
 
-			return generate( prompt, prevGeneratedIds, abortController.current.signal )
+			const requestBody = {
+				prompt,
+				attachments: attachments.map( ( { type, content, label } ) => {
+					// Send only the data that is needed for the generation.
+					return {
+						type,
+						content,
+						label,
+					};
+				} ),
+				prevGeneratedIds,
```

Comment:
might be stupid, but maybe put the `prevGeneratedIds` above the map (for readbility)?